### DISCUSSION TITLE
Returning value of button click in Modal

### DISCUSSION THREAD

**Discussion by rbureau85** [RocketCommunicationsInc/astro#207] [Q&A] [answered]
Hello,
Another one for ya, I only see one return function for the RuxModal. How do I access which button was clicked in the modal?

Thanks :-)

**Comment by markacianfrani** [comment, marked as answer]
You should be able to listen in on the ruxmodalclosed event, which will return false if the user canceled or true if they accepted.

Something like:
```
   <RuxModal
        open
        onRuxmodalclosed={(e) => {
          // e.detail contains the data returned by the custom event
          if (e.detail) {
            console.log("User confirmed!");
          } else {
            console.log("User canceled");
          }
        }}
      />
```
https://codesandbox.io/s/condescending-breeze-7c2dy?file=/src/App.js

Does that work for you? 

We're also reworking Modal in 7.0 to use slots to give you more control over each piece of the Modal, but unfortunately that'll require a breaking change so we can't ship it immediately.

**Reply by rbureau85** [reply]
Great! thank you, I had tried this in the storybook but I guess it had a hiccup cause it showed detail: true on both buttons. Probably just a cache error
